ambry-store/src/main/java/com.github.ambry.store/PersistentIndex.java

```diff
@@ -1068,7 +1179,7 @@ FindInfo findDeletedEntriesSince(FindToken token, long maxTotalSizeOfEntries, lo
         // Find the index segment corresponding to the token indexStartOffset.
         // Get entries starting from the token Key in this index.
         newToken = findEntriesFromSegmentStartOffset(storeToken.getOffset(), storeToken.getStoreKey(), messageEntries,
-            new FindEntriesCondition(maxTotalSizeOfEntries, endTimeSeconds));
+            new FindEntriesCondition(maxTotalSizeOfEntries, endTimeSeconds), indexesRef.get());
```

Comment:
minor. You could get the snapshot of indexes before "if else block" so that you can use it in either of both.